bridge/slack/slack.go

```diff
@@ -311,6 +318,48 @@ func (b *Bslack) sendRTM(msg config.Message) (string, error) {
 	return b.postMessage(&msg, messageParameters, channelInfo)
 }
 
+func (b *Bslack) updateTopicOrPurpose(msg *config.Message, channelInfo *slack.Channel) (bool, error) {
+	var updateFunc func(channelID string, value string) (*slack.Channel, error)
+
+	incomingChangeType, text := b.extractTopicOrPurpose(msg.Text)
+	switch incomingChangeType {
+	case "topic":
+		updateFunc = b.rtm.SetTopicOfConversation
+	case "purpose":
+		updateFunc = b.rtm.SetPurposeOfConversation
+	default:
+		b.Log.Errorf("Unhandled type received from extractTopicOrPurpose: %s", incomingChangeType)
+	}
+	for {
+		_, err := updateFunc(channelInfo.ID, text)
+		if err == nil {
+			return true, nil
+		}
+		if err = b.handleRateLimit(err); err != nil {
+			return true, err
+		}
+	}
+}
+
+// handles updating topic/purpose and determining whether to further propagate update messages
+func (b *Bslack) handleTopicOrPurpose(msg *config.Message, channelInfo *slack.Channel) (bool, error) {
+	if msg.Event != config.EVENT_TOPIC_CHANGE {
+		return false, nil
+	}
+
+	if b.GetBool("SyncTopic") {
+		return b.updateTopicOrPurpose(msg, channelInfo)
+	}
+
+	// Pass along to normal message handlers
```

Comment:
I am not entirely convinced that it would make sense to send messages about a topic change on another bridge if it is not being reflected on the actual channel's topic. What's your use case?

Reply:
If we don't do this, there'll be an inconsistent edge-case on Slack, so that when we didn't `SyncTopic`, but had `ShowTopicChange` enabled, messages would essentially get skipped.

Reply:
> I am not entirely convinced that it would make sense to send messages about a topic change on another bridge if it is not being reflected on the actual channel's topic. What's your use case?

To come back to this: It's still important to share the context, even if I have sync disabled. For instance, I have translation, and on one side they keep a japanese topic. I don't want to clobber that, so I will disable sync, but still want to update to go through -- if I add a link, people can choose to update their own topic to suit their needs.

tl;dr - Difference between alerting of information change, and forcing it. Depends on the fuzzy relationships between each side of bridge :)

Reply:
Does this make sense? (It's the final hold-out, I think, but I don't see any change making sense, as we'd lose functionality.)

It's essentially skip logic, because if this bridge is not set to sync, but IS set to display messages, then we want to let message through to the regular message handling. Dropping this would lead to unexpected behaviour imho, and moving it elsewhere would seem to fragment the logic between places. Open to suggestions, but this feels like the right way